src/impl/AudioWorkletNode.ts

```diff
@@ -0,0 +1,101 @@
+import { AudioWorkletOptions, getWorkletProcessor } from './AudioWorklet';
+import { AudioBuffer, BaseAudioContext, ScriptProcessorNode } from '../api';
+import type { MessagePort } from 'worker_threads';
+import { makeMessageChannel, setNextPort } from './AudioWorkletPort';
+import AudioWorkletProcessor from './AudioWorkletProcessor';
+import AudioParam from '../api/AudioParam';
+import AudioParamImpl from './AudioParam';
+
+function audioBufferToArray(audioBuffer: AudioBuffer): Float32Array[] {
+  const out = [];
+  for (let i = 0; i < audioBuffer.numberOfChannels; i++) {
+    out.push(audioBuffer.getChannelData(i));
+  }
+  return out;
+}
+
+interface WorkletScriptProcessorNode extends ScriptProcessorNode {
+  parameters: Map<string, AudioParam>;
+  port: MessagePort;
+  processor: typeof AudioWorkletProcessor;
+  instance: AudioWorkletProcessor;
+}
+
+export default function AudioWorkletNode(
+  context: BaseAudioContext,
+  name: string,
+  options: Partial<AudioWorkletOptions> = {},
+) {
+  const Processor = getWorkletProcessor(context, name);
+  if (!Processor) {
+    throw new Error(`Cannot initialize unregistered worklet: ${name}`);
+  }
+  const outputChannels = options.outputChannelCount?.[0] ?? 2;
+  // FIXME: should this define number of input channels?
+  const scriptProcessor = context.createScriptProcessor(
+    128, // to match the current impl of AudioWorkletProcessor in browsers
```

Comment:
I usually like to document standards-based things with a reference to where it is defined

```suggestion
    128, // to match the current impl of AudioWorkletProcessor in browsers https://webaudio.github.io/web-audio-api/#rendering-loop
```